### PR TITLE
[CI] Bump tensorflow version to 2.9.1

### PR DESCRIPTION
**Pull request by JiahaoYao**
it seems that the tensorflow version has conflicts with other python package in `Ray Air` development. https://github.com/ray-project/ray/pull/24895

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by JiahaoYao** [comment]
image

Hi @Yard1 , do u know what might cause this? the mac version is too old?

**Comment by JiahaoYao** [comment]
@amogkam do u know how to successfully bump up the tensorflow version? (one reason is that it seems conflict with the jax version on small package.)

File: python/requirements/ml/requirements_dl.txt [hunk 1]
```diff
@@ -1,8 +1,11 @@
 # These requirements are used for the CI and CPU-only Docker images so we install CPU only versions of torch.
 # For GPU Docker images, you should install requirements_ml_docker.txt afterwards.
 
-tensorflow==2.6.2
-tensorflow-probability==0.14.1
+tensorflow==2.6.2;sys_platform=="darwin"
+tensorflow-probability==0.14.1;sys_platform=="darwin"
+
+tensorflow==2.9.1;sys_platform!="darwin"
+tensorflow-probability==0.16.0;sys_platform!="darwin"
```
[the remark this answers is not in the document]
Comment (JiahaoYao) [reply python/requirements/ml/requirements_dl.txt:8]: @ddelange  agreed, if u see this, the below images seems to only support tensorflow 2.6.* on the mac? 

> image
> 
> Hi @Yard1 , do u know what might cause this? the mac version is too old?

[the remark this answers is not in the document]
Comment (ddelange) [reply python/requirements/ml/requirements_dl.txt:8]: from the second line, it looks like this is python 3.6, which I could imagine they stopped supporting, unlike ray. Would be evident in `setup.py` under `python_requires`. See also last bullet point here https://github.com/ray-project/ray/pull/24895#discussion_r905924644

**Comment by JiahaoYao** [comment]
reopen

File: python/requirements/ml/requirements_dl.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (JiahaoYao) [reply python/requirements/ml/requirements_dl.txt]: yes, you are right @ddelange

**Pull request closed without merging**